File: suc/kotlin/004.kt
package com.example.singlefileui

import android.os.Bundle
import androidx.activity.ComponentActivity
import androidx.activity.compose.setContent
import androidx.compose.foundation.BorderStroke
import androidx.compose.foundation.background
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.PaddingValues
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.WindowInsets
import androidx.compose.foundation.layout.aspectRatio
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.heightIn
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.layout.width
import androidx.compose.foundation.lazy.LazyColumn
import androidx.compose.foundation.lazy.items
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.AssistChip
import androidx.compose.material3.AssistChipDefaults
import androidx.compose.material3.Button
import androidx.compose.material3.ButtonDefaults
import androidx.compose.material3.Card
import androidx.compose.material3.CardDefaults
import androidx.compose.material3.CenterAlignedTopAppBar
import androidx.compose.material3.ExperimentalMaterial3Api
import androidx.compose.material3.FilterChip
import androidx.compose.material3.FilterChipDefaults
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Scaffold
import androidx.compose.material3.Shapes
import androidx.compose.material3.Surface
import androidx.compose.material3.Switch
import androidx.compose.material3.SwitchDefaults
import androidx.compose.material3.Text
import androidx.compose.material3.Typography
import androidx.compose.material3.lightColorScheme
import androidx.compose.runtime.Composable
import androidx.compose.runtime.mutableStateListOf
import androidx.compose.runtime.mutableStateOf
import androidx.compose.runtime.remember
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.TextStyle
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.tooling.preview.Preview
import androidx.compose.ui.unit.Dp
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import androidx.core.view.WindowCompat
import androidx.core.view.WindowInsetsCompat
import androidx.core.view.WindowInsetsControllerCompat

private const val NAME = "007_FinanceWallect_en"
private const val UI_TYPE = "Finance Wallet"
private const val STYLE_THEME = "High-Contrast Accessible"
private const val LANG = "en"
private const val BASELINE_SIZE = "1280x720"

object AppTokens {
    object Colors {
        val primary = Color(0xFF1F2937)
        val secondary = Color(0xFF10B981)
        val tertiary = Color(0xFF3B82F6)
        val background = Color(0xFFF3F4F6)
        val surface = Color(0xFFFFFFFF)
        val surfaceVariant = Color(0xFFE5E7EB)
        val outline = Color(0xFFD1D5DB)
        val success = Color(0xFF16A34A)
        val warning = Color(0xFFF59E0B)
        val error = Color(0xFFEF4444)
        val onPrimary = Color(0xFFFFFFFF)
        val onSecondary = Color(0xFF0B1220)
        val onTertiary = Color(0xFFFFFFFF)
        val onBackground = Color(0xFF0B1220)
        val onSurface = Color(0xFF0B1220)
    }
    object TypographyTokens {
        val display = TextStyle(fontSize = 24.sp, fontWeight = FontWeight.SemiBold, lineHeight = 30.sp, letterSpacing = 0.sp)
        val headline = TextStyle(fontSize = 18.sp, fontWeight = FontWeight.SemiBold, lineHeight = 24.sp, letterSpacing = 0.sp)
        val title = TextStyle(fontSize = 14.sp, fontWeight = FontWeight.Medium, lineHeight = 18.sp, letterSpacing = 0.sp)
        val body = TextStyle(fontSize = 12.sp, fontWeight = FontWeight.Normal, lineHeight = 16.sp, letterSpacing = 0.sp)
        val label = TextStyle(fontSize = 10.sp, fontWeight = FontWeight.Medium, lineHeight = 14.sp, letterSpacing = 0.sp)
    }
    object Shapes {
        val small = RoundedCornerShape(6.dp)
        val medium = RoundedCornerShape(8.dp)
        val large = RoundedCornerShape(12.dp)
    }
    object Spacing {
        val xs = 2.dp
        val sm = 4.dp
        val md = 6.dp
        val lg = 8.dp
        val xl = 12.dp
        val xxl = 16.dp
        val xxxl = 24.dp
    }
    data class ShadowSpec(val elevation: Dp, val radius: Dp, val dy: Dp, val opacity: Float)
    object ElevationMapping {
        val level1 = ShadowSpec(1.dp, 2.dp, 0.5.dp, 0.12f)
        val level2 = ShadowSpec(2.dp, 4.dp, 1.dp, 0.14f)
        val level3 = ShadowSpec(3.dp, 6.dp, 1.5.dp, 0.16f)
        val level4 = ShadowSpec(4.dp, 8.dp, 2.dp, 0.18f)
        val level5 = ShadowSpec(5.dp, 10.dp, 2.5.dp, 0.2f)
    }
}

private val AppColorScheme = lightColorScheme(
    primary = AppTokens.Colors.primary,
    onPrimary = AppTokens.Colors.onPrimary,
    secondary = AppTokens.Colors.secondary,
    onSecondary = AppTokens.Colors.onSecondary,
    tertiary = AppTokens.Colors.tertiary,
    onTertiary = AppTokens.Colors.onTertiary,
    background = AppTokens.Colors.background,
    onBackground = AppTokens.Colors.onBackground,
    surface = AppTokens.Colors.surface,
    onSurface = AppTokens.Colors.onSurface,
    surfaceVariant = AppTokens.Colors.surfaceVariant,
    outline = AppTokens.Colors.outline,
    error = AppTokens.Colors.error
)

private val AppTypography = Typography(
    displayLarge = AppTokens.TypographyTokens.display,
    headlineMedium = AppTokens.TypographyTokens.headline,
    titleMedium = AppTokens.TypographyTokens.title,
    bodyMedium = AppTokens.TypographyTokens.body,
    labelMedium = AppTokens.TypographyTokens.label
)

@Composable
fun AppTheme(content: @Composable () -> Unit) {
    MaterialTheme(
        colorScheme = AppColorScheme,
        typography = AppTypography,
        shapes = Shapes(
            extraSmall = AppTokens.Shapes.small,
            small = AppTokens.Shapes.small,
            medium = AppTokens.Shapes.medium,
            large = AppTokens.Shapes.large,
            extraLarge = AppTokens.Shapes.large
        ),
        content = content
    )
}

data class AccountCard(val id: Int, val name: String, val balance: String, val accent: Color)
data class TxItem(val id: Int, val title: String, val subtitle: String, val amount: String, val positive: Boolean)

@OptIn(ExperimentalMaterial3Api::class)
@Composable
fun RootScreen() {
    val showBalance = remember { mutableStateOf(true) }
    val range = listOf("All", "Week", "Month", "Year")
    val selected = remember { mutableStateOf("Month") }
    val flags = remember { mutableStateListOf<String>() }
    val accounts = remember {
        listOf(
            AccountCard(1, "Wallet", "$2,845.20", Color(0xFF10B981)),
            AccountCard(2, "Savings", "$8,120.00", Color(0xFF3B82F6)),
            AccountCard(3, "Investments", "$15,430.55", Color(0xFFF59E0B))
        )
    }
    val txs = remember {
        listOf(
            TxItem(1, "Grocery Market", "Today • 12:40", "-$32.18", false),
            TxItem(2, "Salary", "Yesterday • 09:00", "+$3,200.00", true),
            TxItem(3, "Coffee Shop", "Yesterday • 15:22", "-$4.90", false),
            TxItem(4, "ETF Purchase", "Mon • 10:05", "-$250.00", false),
            TxItem(5, "Refund", "Sun • 18:40", "+$18.99", true)
        )
    }
    Scaffold(
        contentWindowInsets = WindowInsets(0),
        topBar = {
            CenterAlignedTopAppBar(
                title = {
                    Row(verticalAlignment = Alignment.CenterVertically, horizontalArrangement = Arrangement.spacedBy(AppTokens.Spacing.sm)) {
                        Box(modifier = Modifier.size(20.dp).background(AppTokens.Colors.primary, AppTokens.Shapes.small))
                        Text(text = "Wallet", style = MaterialTheme.typography.displayLarge, color = MaterialTheme.colorScheme.onSurface)
                    }
                }
            )
        },
        bottomBar = {
            Surface(color = MaterialTheme.colorScheme.surface, tonalElevation = AppTokens.ElevationMapping.level3.elevation, shadowElevation = AppTokens.ElevationMapping.level3.elevation) {
                Row(modifier = Modifier.fillMaxWidth().padding(horizontal = AppTokens.Spacing.lg, vertical = AppTokens.Spacing.md), horizontalArrangement = Arrangement.SpaceBetween, verticalAlignment = Alignment.CenterVertically) {
                    Column(verticalArrangement = Arrangement.spacedBy(AppTokens.Spacing.xs), horizontalAlignment = Alignment.Start) {
                        Text(text = "Total Balance", style = MaterialTheme.typography.labelMedium, color = MaterialTheme.colorScheme.onSurface)
                        Text(text = if (showBalance.value) "$26,395.75" else "•••••••", style = MaterialTheme.typography.headlineMedium, color = MaterialTheme.colorScheme.onSurface)
                    }
                    Button(onClick = {}, colors = ButtonDefaults.buttonColors(containerColor = MaterialTheme.colorScheme.secondary, contentColor = MaterialTheme.colorScheme.onSecondary), shape = AppTokens.Shapes.large, modifier = Modifier.height(44.dp).width(140.dp)) {
                        Text(text = "Add Money", style = MaterialTheme.typography.titleMedium)
                    }
                }
            }
        },
        containerColor = MaterialTheme.colorScheme.background
    ) { padding ->
        LazyColumn(modifier = Modifier.fillMaxSize().padding(padding).padding(horizontal = AppTokens.Spacing.lg, vertical = AppTokens.Spacing.md), verticalArrangement = Arrangement.spacedBy(AppTokens.Spacing.lg), contentPadding = PaddingValues(bottom = AppTokens.Spacing.xxxl)) {
            item {
                Row(modifier = Modifier.fillMaxWidth(), horizontalArrangement = Arrangement.SpaceBetween, verticalAlignment = Alignment.CenterVertically) {
                    Row(horizontalArrangement = Arrangement.spacedBy(AppTokens.Spacing.sm), verticalAlignment = Alignment.CenterVertically) {
                        range.forEach { r ->
                            val active = r == selected.value
                            FilterChip(selected = active, onClick = { selected.value = r }, label = { Text(text = r, style = if (active) MaterialTheme.typography.titleMedium else MaterialTheme.typography.bodyMedium, color = if (active) MaterialTheme.colorScheme.onPrimary else MaterialTheme.colorScheme.onSurface) }, colors = FilterChipDefaults.filterChipColors(containerColor = if (active) MaterialTheme.colorScheme.primary else MaterialTheme.colorScheme.surface, selectedContainerColor = MaterialTheme.colorScheme.primary), shape = AppTokens.Shapes.medium, border = BorderStroke(1.dp, MaterialTheme.colorScheme.outline))
                        }
                    }
                    Row(horizontalArrangement = Arrangement.spacedBy(AppTokens.Spacing.sm), verticalAlignment = Alignment.CenterVertically) {
                        Text(text = "Hide", style = MaterialTheme.typography.labelMedium, color = MaterialTheme.colorScheme.onSurface)
                        Switch(checked = !showBalance.value, onCheckedChange = { showBalance.value = !it }, colors = SwitchDefaults.colors(checkedThumbColor = MaterialTheme.colorScheme.onPrimary, checkedTrackColor = MaterialTheme.colorScheme.primary))
                    }
                }
            }
            items(accounts) { a ->
                Card(shape = AppTokens.Shapes.large, colors = CardDefaults.cardColors(containerColor = MaterialTheme.colorScheme.surface), elevation = CardDefaults.cardElevation(defaultElevation = AppTokens.ElevationMapping.level2.elevation), border = BorderStroke(1.dp, MaterialTheme.colorScheme.outline), modifier = Modifier.fillMaxWidth().heightIn(min = 120.dp)) {
                    Column(modifier = Modifier.fillMaxWidth().padding(AppTokens.Spacing.lg), verticalArrangement = Arrangement.spacedBy(AppTokens.Spacing.sm), horizontalAlignment = Alignment.Start) {
                        Row(modifier = Modifier.fillMaxWidth(), horizontalArrangement = Arrangement.SpaceBetween, verticalAlignment = Alignment.CenterVertically) {
                            Row(verticalAlignment = Alignment.CenterVertically, horizontalArrangement = Arrangement.spacedBy(AppTokens.Spacing.sm)) {
                                Box(modifier = Modifier.size(16.dp).background(a.accent, AppTokens.Shapes.small))
                                Text(text = a.name, style = MaterialTheme.typography.titleMedium, color = MaterialTheme.colorScheme.onSurface)
                            }
                            AssistChip(onClick = {}, label = { Text(text = "Manage", style = MaterialTheme.typography.labelMedium, color = MaterialTheme.colorScheme.onSecondary) }, shape = AppTokens.Shapes.small, border = BorderStroke(1.dp, MaterialTheme.colorScheme.outline), colors = AssistChipDefaults.assistChipColors(containerColor = MaterialTheme.colorScheme.secondary))
                        }
                        Box(modifier = Modifier.fillMaxWidth().aspectRatio(2.8f).background(AppTokens.Colors.surfaceVariant, AppTokens.Shapes.medium), contentAlignment = Alignment.Center) {
                            Text(text = if (showBalance.value) a.balance else "•••••••", style = MaterialTheme.typography.headlineMedium, color = MaterialTheme.colorScheme.onSurface)
                        }
                        Row(horizontalArrangement = Arrangement.spacedBy(AppTokens.Spacing.sm), verticalAlignment = Alignment.CenterVertically) {
                            AssistChip(onClick = {}, label = { Text(text = "Transfer", style = MaterialTheme.typography.labelMedium, color = MaterialTheme.colorScheme.onSecondary) }, shape = AppTokens.Shapes.small, border = BorderStroke(1.dp, MaterialTheme.colorScheme.outline), colors = AssistChipDefaults.assistChipColors(containerColor = MaterialTheme.colorScheme.secondary))
                            AssistChip(onClick = {}, label = { Text(text = "Details", style = MaterialTheme.typography.labelMedium, color = MaterialTheme.colorScheme.onSurface) }, shape = AppTokens.Shapes.small, border = BorderStroke(1.dp, MaterialTheme.colorScheme.outline), colors = AssistChipDefaults.assistChipColors(containerColor = MaterialTheme.colorScheme.surface))
                        }
                    }
                }
            }
            item {
                Row(modifier = Modifier.fillMaxWidth(), horizontalArrangement = Arrangement.SpaceBetween, verticalAlignment = Alignment.CenterVertically) {
                    Text(text = "Recent Activity", style = MaterialTheme.typography.headlineMedium, color = MaterialTheme.colorScheme.onSurface)
                    Row(horizontalArrangement = Arrangement.spacedBy(AppTokens.Spacing.sm), verticalAlignment = Alignment.CenterVertically) {
                        AssistChip(onClick = { if (flags.contains("Income")) flags.remove("Income") else flags.add("Income") }, label = { Text(text = "Income", style = MaterialTheme.typography.labelMedium, color = if (flags.contains("Income")) MaterialTheme.colorScheme.onSecondary else MaterialTheme.colorScheme.onSurface) }, shape = AppTokens.Shapes.small, border = BorderStroke(1.dp, MaterialTheme.colorScheme.outline), colors = AssistChipDefaults.assistChipColors(containerColor = if (flags.contains("Income")) MaterialTheme.colorScheme.secondary else MaterialTheme.colorScheme.surface))
                        AssistChip(onClick = { if (flags.contains("Expense")) flags.remove("Expense") else flags.add("Expense") }, label = { Text(text = "Expense", style = MaterialTheme.typography.labelMedium, color = if (flags.contains("Expense")) MaterialTheme.colorScheme.onSecondary else MaterialTheme.colorScheme.onSurface) }, shape = AppTokens.Shapes.small, border = BorderStroke(1.dp, MaterialTheme.colorScheme.outline), colors = AssistChipDefaults.assistChipColors(containerColor = if (flags.contains("Expense")) MaterialTheme.colorScheme.secondary else MaterialTheme.colorScheme.surface))
                    }
                }
            }
            items(txs) { t ->
                Card(shape = AppTokens.Shapes.large, colors = CardDefaults.cardColors(containerColor = MaterialTheme.colorScheme.surface), elevation = CardDefaults.cardElevation(defaultElevation = AppTokens.ElevationMapping.level1.elevation), border = BorderStroke(1.dp, MaterialTheme.colorScheme.outline), modifier = Modifier.fillMaxWidth().heightIn(min = 72.dp)) {
                    Row(modifier = Modifier.fillMaxWidth().padding(AppTokens.Spacing.lg), horizontalArrangement = Arrangement.SpaceBetween, verticalAlignment = Alignment.CenterVertically) {
                        Row(verticalAlignment = Alignment.CenterVertically, horizontalArrangement = Arrangement.spacedBy(AppTokens.Spacing.md)) {
                            Box(modifier = Modifier.size(32.dp).background(if (t.positive) AppTokens.Colors.success else AppTokens.Colors.error, AppTokens.Shapes.small))
                            Column(verticalArrangement = Arrangement.spacedBy(AppTokens.Spacing.xs), horizontalAlignment = Alignment.Start) {
                                Text(text = t.title, style = MaterialTheme.typography.titleMedium, color = MaterialTheme.colorScheme.onSurface)
                                Text(text = t.subtitle, style = MaterialTheme.typography.bodyMedium, color = MaterialTheme.colorScheme.onSurface)
                            }
                        }
                        Text(text = t.amount, style = MaterialTheme.typography.titleMedium, color = if (t.positive) AppTokens.Colors.success else AppTokens.Colors.error)
                    }
                }
            }
            item { Spacer(modifier = Modifier.height(AppTokens.Spacing.xxl)) }
        }
    }
}

class MainActivity : ComponentActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        WindowCompat.setDecorFitsSystemWindows(window, false)
        val controller = WindowInsetsControllerCompat(window, window.decorView)
        controller.systemBarsBehavior = WindowInsetsControllerCompat.BEHAVIOR_SHOW_TRANSIENT_BARS_BY_SWIPE
        controller.hide(WindowInsetsCompat.Type.navigationBars())
        setContent {
            AppTheme {
                Surface(color = MaterialTheme.colorScheme.background) {
                    RootScreen()
                }
            }
        }
    }
    override fun onWindowFocusChanged(hasFocus: Boolean) {
        super.onWindowFocusChanged(hasFocus)
        if (hasFocus) {
            val controller = WindowInsetsControllerCompat(window, window.decorView)
            controller.systemBarsBehavior = WindowInsetsControllerCompat.BEHAVIOR_SHOW_TRANSIENT_BARS_BY_SWIPE
            controller.hide(WindowInsetsCompat.Type.navigationBars())
        }
    }
}

@Preview(showBackground = true, backgroundColor = 0xFFFFFFFF, widthDp = 360, heightDp = 640)
@Composable
fun PreviewRoot() {
    AppTheme {
        RootScreen()
    }
}

File: suc/swift/004.swift
import SwiftUI

// MARK: - Entry Point
@main
struct FinanceWalletApp: App {
    var body: some Scene {
        WindowGroup {
            ContentView()
        }
    }
}

// MARK: - App Tokens & Colors
struct AppTokens {
    struct Colors {
        static let primary = Color(hex: 0x1F2937)
        static let secondary = Color(hex: 0x10B981)
        static let tertiary = Color(hex: 0x3B82F6)
        static let background = Color(hex: 0xF3F4F6)
        static let surface = Color(hex: 0xFFFFFFFF)
        static let surfaceVariant = Color(hex: 0xE5E7EB)
        static let outline = Color(hex: 0xD1D5DB)
        static let success = Color(hex: 0x16A34A)
        static let warning = Color(hex: 0xF59E0B)
        static let error = Color(hex: 0xEF4444)
        static let onPrimary = Color(hex: 0xFFFFFFFF)
        static let onSecondary = Color(hex: 0x0B1220)
        static let onSurface = Color(hex: 0x0B1220)
    }
    
    struct Typography {
        static let display = Font.system(size: 24, weight: .semibold)
        static let headline = Font.system(size: 18, weight: .semibold)
        static let title = Font.system(size: 14, weight: .medium)
        static let body = Font.system(size: 12, weight: .regular)
        static let label = Font.system(size: 10, weight: .medium)
    }
    
    struct Spacing {
        static let xs: CGFloat = 2
        static let sm: CGFloat = 4
        static let md: CGFloat = 6
        static let lg: CGFloat = 8
        static let xl: CGFloat = 12
        static let xxl: CGFloat = 16
        static let xxxl: CGFloat = 24
    }
    
    struct Shapes {
        static let small: CGFloat = 6
        static let medium: CGFloat = 8
        static let large: CGFloat = 12
    }
}

// MARK: - Models
struct AccountCardModel: Identifiable {
    let id: Int
    let name: String
    let balance: String
    let accent: Color
}

struct TxItemModel: Identifiable {
    let id: Int
    let title: String
    let subtitle: String
    let amount: String
    let positive: BooleanLiteralType
}

// MARK: - Root View
struct ContentView: View {
    // State
    @State private var showBalance = true
    @State private var selectedFilter = "Month"
    @State private var flags: Set<String> = []
    
    // Data
    let filters = ["All", "Week", "Month", "Year"]
    let accounts = [
        AccountCardModel(id: 1, name: "Wallet", balance: "$2,845.20", accent: Color(hex: 0x10B981)),
        AccountCardModel(id: 2, name: "Savings", balance: "$8,120.00", accent: Color(hex: 0x3B82F6)),
        AccountCardModel(id: 3, name: "Investments", balance: "$15,430.55", accent: Color(hex: 0xF59E0B))
    ]
    let transactions = [
        TxItemModel(id: 1, title: "Grocery Market", subtitle: "Today • 12:40", amount: "-$32.18", positive: false),
        TxItemModel(id: 2, title: "Salary", subtitle: "Yesterday • 09:00", amount: "+$3,200.00", positive: true),
        TxItemModel(id: 3, title: "Coffee Shop", subtitle: "Yesterday • 15:22", amount: "-$4.90", positive: false),
        TxItemModel(id: 4, title: "ETF Purchase", subtitle: "Mon • 10:05", amount: "-$250.00", positive: false),
        TxItemModel(id: 5, title: "Refund", subtitle: "Sun • 18:40", amount: "+$18.99", positive: true)
    ]
    
    var body: some View {
        ZStack {
            // Background
            AppTokens.Colors.background.ignoresSafeArea()
            
            VStack(spacing: 0) {
                // Top Bar
                HStack(spacing: AppTokens.Spacing.sm) {
                    RoundedRectangle(cornerRadius: AppTokens.Shapes.small)
                        .fill(AppTokens.Colors.primary)
                        .frame(width: 20, height: 20)
                    Text("Wallet")
                        .font(AppTokens.Typography.display)
                        .foregroundColor(AppTokens.Colors.onSurface)
                    Spacer()
                }
                .padding()
                .background(AppTokens.Colors.background) // Match background for sticky feel
                
                // Content
                ScrollView(showsIndicators: false) {
                    VStack(spacing: AppTokens.Spacing.lg) {
                        
                        // Filters Row
                        HStack {
                            HStack(spacing: AppTokens.Spacing.sm) {
                                ForEach(filters, id: \.self) { filter in
                                    FilterChip(
                                        label: filter,
                                        isSelected: selectedFilter == filter,
                                        onClick: { selectedFilter = filter }
                                    )
                                }
                            }
                            
                            Spacer()
                            
                            HStack(spacing: AppTokens.Spacing.sm) {
                                Text("Hide")
                                    .font(AppTokens.Typography.label)
                                    .foregroundColor(AppTokens.Colors.onSurface)
                                
                                Toggle("", isOn: Binding(
                                    get: { !showBalance },
                                    set: { showBalance = !$0 }
                                ))
                                .labelsHidden()
                                .toggleStyle(CustomSwitchStyle())
                                .frame(height: 24)
                            }
                        }
                        
                        // Accounts List
                        ForEach(accounts) { account in
                            AccountCardView(account: account, showBalance: showBalance)
                        }
                        
                        // Recent Activity Header
                        HStack {
                            Text("Recent Activity")
                                .font(AppTokens.Typography.headline)
                                .foregroundColor(AppTokens.Colors.onSurface)
                            
                            Spacer()
                            
                            HStack(spacing: AppTokens.Spacing.sm) {
                                AssistChip(
                                    label: "Income",
                                    isSelected: flags.contains("Income"),
                                    onClick: {
                                        if flags.contains("Income") { flags.remove("Income") }
                                        else { flags.insert("Income") }
                                    }
                                )
                                AssistChip(
                                    label: "Expense",
                                    isSelected: flags.contains("Expense"),
                                    onClick: {
                                        if flags.contains("Expense") { flags.remove("Expense") }
                                        else { flags.insert("Expense") }
                                    }
                                )
                            }
                        }
                        
                        // Transactions List
                        ForEach(transactions) { tx in
                            TransactionCard(tx: tx)
                        }
                        
                        // Bottom Padding for FAB/BottomBar
                        Spacer().frame(height: 100)
                    }
                    .padding(.horizontal, AppTokens.Spacing.lg)
                    .padding(.vertical, AppTokens.Spacing.md)
                }
            }
            
            // Bottom Bar Overlay
            VStack {
                Spacer()
                
                VStack {
                    HStack(alignment: .center) {
                        VStack(alignment: .leading, spacing: AppTokens.Spacing.xs) {
                            Text("Total Balance")
                                .font(AppTokens.Typography.title) // labelMedium in Kotlin, upped slightly for visibility
                                .foregroundColor(AppTokens.Colors.onSurface)
                            Text(showBalance ? "$26,395.75" : "•••••••")
                                .font(AppTokens.Typography.headline)
                                .foregroundColor(AppTokens.Colors.onSurface)
                        }
                        
                        Spacer()
                        
                        Button(action: {}) {
                            Text("Add Money")
                                .font(AppTokens.Typography.title)
                                .foregroundColor(AppTokens.Colors.onSecondary)
                                .frame(width: 140, height: 44)
                                .background(AppTokens.Colors.secondary)
                                .cornerRadius(AppTokens.Shapes.large)
                        }
                    }
                    .padding(.horizontal, AppTokens.Spacing.lg)
                    .padding(.vertical, AppTokens.Spacing.md)
                }
                .background(AppTokens.Colors.surface)
                .shadow(color: Color.black.opacity(0.16), radius: 6, x: 0, y: -1.5) // Level 3 Shadow approx
            }
        }
        .statusBarHidden(true)
    }
}

// MARK: - Components

struct FilterChip: View {
    let label: String
    let isSelected: Bool
    let onClick: () -> Void
    
    var body: some View {
        Button(action: onClick) {
            Text(label)
                .font(isSelected ? AppTokens.Typography.title : AppTokens.Typography.body)
                .foregroundColor(isSelected ? AppTokens.Colors.onPrimary : AppTokens.Colors.onSurface)
                .padding(.horizontal, 12)
                .padding(.vertical, 8)
                .background(isSelected ? AppTokens.Colors.primary : AppTokens.Colors.surface)
                .cornerRadius(AppTokens.Shapes.medium)
                .overlay(
                    RoundedRectangle(cornerRadius: AppTokens.Shapes.medium)
                        .stroke(AppTokens.Colors.outline, lineWidth: isSelected ? 0 : 1)
                )
        }
    }
}

struct AssistChip: View {
    let label: String
    let isSelected: Bool
    let onClick: () -> Void
    
    var body: some View {
        Button(action: onClick) {
            Text(label)
                .font(AppTokens.Typography.label)
                .foregroundColor(isSelected ? AppTokens.Colors.onSecondary : AppTokens.Colors.onSurface)
                .padding(.horizontal, 8)
                .padding(.vertical, 6)
                .background(isSelected ? AppTokens.Colors.secondary : AppTokens.Colors.surface)
                .cornerRadius(AppTokens.Shapes.small)
                .overlay(
                    RoundedRectangle(cornerRadius: AppTokens.Shapes.small)
                        .stroke(AppTokens.Colors.outline, lineWidth: 1)
                )
        }
    }
}

struct AccountCardView: View {
    let account: AccountCardModel
    let showBalance: Bool
    
    var body: some View {
        VStack(alignment: .leading, spacing: AppTokens.Spacing.sm) {
            // Header
            HStack {
                HStack(spacing: AppTokens.Spacing.sm) {
                    RoundedRectangle(cornerRadius: AppTokens.Shapes.small)
                        .fill(account.accent)
                        .frame(width: 16, height: 16)
                    Text(account.name)
                        .font(AppTokens.Typography.title)
                        .foregroundColor(AppTokens.Colors.onSurface)
                }
                Spacer()
                Button(action: {}) {
                    Text("Manage")
                        .font(AppTokens.Typography.label)
                        .foregroundColor(AppTokens.Colors.onSecondary)
                        .padding(.horizontal, 8)
                        .padding(.vertical, 4)
                        .background(AppTokens.Colors.secondary)
                        .cornerRadius(AppTokens.Shapes.small)
                        .overlay(
                            RoundedRectangle(cornerRadius: AppTokens.Shapes.small)
                                .stroke(AppTokens.Colors.outline, lineWidth: 1)
                        )
                }
            }
            
            // Balance Box
            ZStack {
                AppTokens.Colors.surfaceVariant
                Text(showBalance ? account.balance : "•••••••")
                    .font(AppTokens.Typography.headline)
                    .foregroundColor(AppTokens.Colors.onSurface)
            }
            .cornerRadius(AppTokens.Shapes.medium)
            .aspectRatio(2.8, contentMode: .fit)
            .frame(maxWidth: .infinity)
            
            // Actions
            HStack(spacing: AppTokens.Spacing.sm) {
                // Transfer Button
                Button(action: {}) {
                    Text("Transfer")
                        .font(AppTokens.Typography.label)
                        .foregroundColor(AppTokens.Colors.onSecondary)
                        .padding(.horizontal, 8)
                        .padding(.vertical, 6)
                        .background(AppTokens.Colors.secondary)
                        .cornerRadius(AppTokens.Shapes.small)
                        .overlay(
                            RoundedRectangle(cornerRadius: AppTokens.Shapes.small)
                                .stroke(AppTokens.Colors.outline, lineWidth: 1)
                        )
                }
                
                // Details Button
                Button(action: {}) {
                    Text("Details")
                        .font(AppTokens.Typography.label)
                        .foregroundColor(AppTokens.Colors.onSurface)
                        .padding(.horizontal, 8)
                        .padding(.vertical, 6)
                        .background(AppTokens.Colors.surface)
                        .cornerRadius(AppTokens.Shapes.small)
                        .overlay(
                            RoundedRectangle(cornerRadius: AppTokens.Shapes.small)
                                .stroke(AppTokens.Colors.outline, lineWidth: 1)
                        )
                }
            }
        }
        .padding(AppTokens.Spacing.lg)
        .background(AppTokens.Colors.surface)
        .cornerRadius(AppTokens.Shapes.large)
        .shadow(color: Color.black.opacity(0.14), radius: 4, x: 0, y: 1) // Level 2 Shadow
        .overlay(
            RoundedRectangle(cornerRadius: AppTokens.Shapes.large)
                .stroke(AppTokens.Colors.outline, lineWidth: 1)
        )
        .frame(minHeight: 120)
    }
}

struct TransactionCard: View {
    let tx: TxItemModel
    
    var body: some View {
        HStack {
            HStack(spacing: AppTokens.Spacing.md) {
                RoundedRectangle(cornerRadius: AppTokens.Shapes.small)
                    .fill(tx.positive ? AppTokens.Colors.success : AppTokens.Colors.error)
                    .frame(width: 32, height: 32)
                
                VStack(alignment: .leading, spacing: AppTokens.Spacing.xs) {
                    Text(tx.title)
                        .font(AppTokens.Typography.title)
                        .foregroundColor(AppTokens.Colors.onSurface)
                    Text(tx.subtitle)
                        .font(AppTokens.Typography.body)
                        .foregroundColor(AppTokens.Colors.onSurface)
                }
            }
            Spacer()
            Text(tx.amount)
                .font(AppTokens.Typography.title)
                .foregroundColor(tx.positive ? AppTokens.Colors.success : AppTokens.Colors.error)
        }
        .padding(AppTokens.Spacing.lg)
        .frame(minHeight: 72)
        .background(AppTokens.Colors.surface)
        .cornerRadius(AppTokens.Shapes.large)
        .shadow(color: Color.black.opacity(0.12), radius: 2, x: 0, y: 0.5) // Level 1 Shadow
        .overlay(
            RoundedRectangle(cornerRadius: AppTokens.Shapes.large)
                .stroke(AppTokens.Colors.outline, lineWidth: 1)
        )
    }
}

// Custom Toggle Style to match the Android switch look closer
struct CustomSwitchStyle: ToggleStyle {
    func makeBody(configuration: Configuration) -> some View {
        HStack {
            configuration.label
            
            ZStack {
                RoundedRectangle(cornerRadius: 16)
                    .fill(configuration.isOn ? AppTokens.Colors.primary : AppTokens.Colors.surfaceVariant)
                    .frame(width: 36, height: 20) // Android switches are smaller/wider usually
                    .overlay(
                        RoundedRectangle(cornerRadius: 16)
                            .stroke(AppTokens.Colors.outline, lineWidth: configuration.isOn ? 0 : 1)
                    )
                
                Circle()
                    .fill(configuration.isOn ? AppTokens.Colors.onPrimary : AppTokens.Colors.outline)
                    .padding(2)
                    .frame(width: 20, height: 20)
                    .offset(x: configuration.isOn ? 8 : -8)
            }
            .onTapGesture {
                withAnimation(.easeInOut(duration: 0.2)) {
                    configuration.isOn.toggle()
                }
            }
        }
    }
}

// MARK: - Color Hex Extension
extension Color {
    init(hex: UInt, alpha: Double = 1) {
        self.init(
            .sRGB,
            red: Double((hex >> 16) & 0xff) / 255,
            green: Double((hex >> 08) & 0xff) / 255,
            blue: Double((hex >> 00) & 0xff) / 255,
            opacity: alpha
        )
    }
}